### DISCUSSION TITLE
Exporting designs by themselves

### DISCUSSION THREAD

**Discussion by WorkQuoh** [iqm-finland/KQCircuits#105] [Q&A] [answered]
Hi, so I was able to export geometries through the command line to ANSYS HFSS by using ```kqc singularity Chip_1.py```. However, I forgot how I was able to do it yesterday. I was wondering what directory I should have my python file in. Every time I run ```kqc singularity Chip_1.py``` nothing happens (also I forgot the name of my file Chip_1.py is just an example) so I am wondering where the python file should be in order to export the geometry to ANSYS HFSS. 

When I try to export in klayout I run into too many errors, so I would rather just be able to export from the command line.

Thanks in advance.

**Comment by qpavsmi** [comment]
I think you meant to use the `kqc sim Chip_1.py` command here. `kqc singularity` compiles the simulation into a [Singularity image](https://docs.sylabs.io/guides/3.5/user-guide/introduction.html), which doesn't host Ansys as a simulator application. `kqc singularity` command is mostly useful if you're operating on a high-performance computing devices, but if you're using your personal computer then there isn't really use for `kqc singularity` command.

**Reply by WorkQuoh** [reply]
I see. I don't know how then, but I was able to export one of my geometries from the command line to HFSS. I was pretty sure I was able to do this with ```kqc singularity```. Do you by chance know any other methods that would allow me to push only the geometry to HFSS using the command line? Because when I try in klayout it just simply doesn't work.

**Reply by qpavsmi** [reply]
What do you mean when you say it doesn't work in KLayout? What does `Chip_1.py` contain, does it resemble [this simulation script](https://github.com/iqm-finland/KQCircuits/blob/main/klayout_package/python/scripts/simulations/xs1_full_chip_sim.py) for example?

**Reply by WorkQuoh** [reply]
So for example, if I have a geometry in klayout. Then if I go to the macro development to export the geometry with this code: ```https://github.com/iqm-finland/KQCircuits/blob/main//klayout_package/python/scripts/macros/export/export_ansys.lym```

I get this error: ```export_ansys_json() got an unexpected keyword argument 'ansys_tool'```

**Reply by qpavsmi** [reply, marked as answer]
We already covered this here: https://github.com/iqm-finland/KQCircuits/discussions/102#discussioncomment-10691109

But if you have `Chip_1.py` you shouldn't need that macro. Do you have the geometry of this chip defined there, or in some other python file?

As long as you have following files:
- `klayout_package/python/kqcircuits/chips/my_chip.py` a chip class which defines geometry of the chip
- `klayout_package/python/kqcircuits/simulations/my_chip_sim.py` a simulation class which places the chip on the layout and defines ports for it
- `klayout_package/python/scripts/simulations/my_chip_sim.py` a simulation script which takes the simulation class, sweeps its parameters, and configures ANSYS to simulate it.

Then `kqc sim my_chip_sim.py` should work
